ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DataColumnSidecarRecoveringCustodyImpl.java

```diff
@@ -93,11 +94,12 @@ public DataColumnSidecarRecoveringCustodyImpl(
     this.kzg = kzg;
     this.spec = spec;
     this.dataColumnSidecarPublisher = dataColumnSidecarPublisher;
-    this.custodyGroupCountManager = custodyGroupCountManager;
+    this.custodyGroupCountManagerSupplier = custodyGroupCountManagerSupplier;
     this.recoveryTasks =
         LimitedMap.createSynchronizedNatural(spec.getGenesisSpec().getSlotsPerEpoch());
     this.isSuperNode =
-        new AtomicBoolean(custodyGroupCountManager.getCustodyGroupCount() == groupCount);
+        new AtomicBoolean(
+            custodyGroupCountManagerSupplier.get().getCustodyGroupCount() == groupCount);
```

Comment:
So we use it right in the constructor, so it's a bit confusing as we passed supplier to delay init.
I think we'd better set it to false on var declaring string, we will override it on first usage anyway in `shouldSkipProcessing`

Reply:
Make sense. It will be updated at the first `onSlot` or `onBlock` and I guess that's fine.